src/net/transport.rs

```diff
@@ -303,4 +306,18 @@ mod e2e_tests {
         let res = result.reconstruct();
         assert_eq!(Fp31::try_from(20u128).unwrap(), res[0]);
     }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn three_helpers_http() {
+        let conf = TestConfigBuilder::with_open_ports()
+            .with_disable_https_option(true)
+            .build();
+        test_three_helpers(conf).await;
+    }
+
+    #[tokio::test(flavor = "multi_thread", worker_threads = 2)]
+    async fn three_helpers_https() {
+        let conf = TestConfigBuilder::with_open_ports().build();
+        test_three_helpers(conf).await;
+    }
 }
```

Comment:
I realize that this is tricky, but maybe we should have tests that validate each of the different client certificate configuration styles.  Except maybe the one where the system trust store is used (which would be too variable and hard to get test certificates for).  The code seems solid, but I can't verify that.

Difficulty in testing might be cause to trim some of the CA-based stuff.